File: liqp-core/src/main/java/liqp/nodes/LtEqNode.java
package liqp.nodes;

import liqp.LValue;
import lombok.Getter;

@Getter
public class LtEqNode extends ExpressionNode{

  public LtEqNode(LNode lhs, LNode rhs) {
    super(lhs, rhs);
  }

  @Override
    public Object render(RenderContext context) {

        Object a = lhs.render(context);
        Object b = rhs.render(context);

        return (a instanceof Number) && (b instanceof Number) &&
                super.asNumber(a).doubleValue() <= super.asNumber(b).doubleValue();
    }
}

File: liqp-core/src/main/kotlin/liqp/nodes/LtEqNode.kt
package liqp.nodes

import liqp.ComparisonResult.EQUAL
import liqp.ComparisonResult.LESS
import liqp.node.LNode

class LtEqNode(lhs: LNode, rhs: LNode) : ComparisonNode(lhs, rhs) {
  override val matchingComparisons = setOf(EQUAL, LESS)
}
